jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/generator/Generator.java

```diff
@@ -45,7 +46,7 @@ public Generator(ByteBufferPool bufferPool, boolean useDirectByteBuffers, int ma
         headerGenerator = new HeaderGenerator(bufferPool, useDirectByteBuffers);
         hpackEncoder = new HpackEncoder();
 
-        this.generators = new FrameGenerator[FrameType.values().length];
+        this.generators = new FrameGenerator[FrameType.CONTINUATION.ordinal() + 1];
```

Comment:
This looks fragile, the array length should be calculated.

Reply:
Not sure how?
The array is filled a few lines below, and any error would be triggered immediately.